gym_electric_motor/visualization/motor_dashboard.py

```diff
@@ -1,140 +1,223 @@
-                    - reward: The reward per step is plotted
-                    - action_{i}: The i-th action is plotted. 'action_0' for discrete action space
-                    - mean_reward: The mean episode reward
-                    
-            update_cycle(int): Number after how many steps the plot shall be updated. (default 1000)
-            dark_mode(Bool):  Select a dark background for visualization by setting it to True
+            state_plots('all'/iterable(str)): An iterable of state names to be shown. If 'all' all states will be shown.
+                Default: () (no plotted states)
+            action_plots('all'/iterable(int)): If action_plots='all', all actions will be plotted. If more than one
+                action can be applied on the environment it can be selected by its index.
+                Default: () (no plotted actions).
+            reward_plot(boolean): Select if the current reward is to be plotted. Default: False
+            additional_plots(iterable((TimePlot/EpisodePlot/StepPlot))): Additional already instantiated plots
+                to be shown on the dashboard
+            update_interval(int > 0): Amount of steps after which the plots are updated. Updating each step reduces the
+                performance drastically. Default: 1000
+            step_plot_width(int > 0): Width of the step plots in steps. Default: 10000 steps
+                (1 second for continuously controlled environments / 0.1 second for discretely controlled environments)
+            style(string): Select one of the matplotlib-styles. e.g. "dark-background".
+                Default: None (the already selected style)
         """
-        self._update_cycle = update_cycle
-        self._figure = None
-        self._figure_ep = None
+        # Basic assertions
+        assert type(reward_plot) is bool
+        assert all(isinstance(ap, (TimePlot, EpisodePlot, StepPlot)) for ap in additional_plots)
+        assert type(update_interval) in [int, float]
+        assert update_interval > 0
+        assert type(step_plot_width) in [int, float]
+        assert step_plot_width > 0
+        assert style in plt.style.available or style is None
+
+        super().__init__()
+
+        # Select the matplotlib style
+        if style is not None:
+            plt.style.use(style)
+        # List of the opened figures
+        self._figures = []
+        # The figures to be opened for the step plots, episodic plots and interval plots
+        self._time_plot_figure = None
+        self._episodic_plot_figure = None
+        self._interval_plot_figure = None
+
+        # Store the input data
+        self._state_plots = state_plots
+        self._action_plots = action_plots
+        self._reward_plot = reward_plot
+
+        # Separate the additional plots into StepPlots, EpisodicPlots and IntervalPlots
+        self._custom_step_plots = list(filter(lambda plot: isinstance(plot, TimePlot), list(additional_plots)))
```

Comment:
```suggestion
        self._custom_step_plots = [p for p in additional_plots if isinstance(p, TimePlot)]
```

the filter function does not seem necessary